src/services/positionService.ts

```diff
@@ -742,6 +834,35 @@ export default class PositionService {
     return this.safeService.submitMultipleTxs([approveTx, depositTx]);
   }
 
+  async xCallDeposit(
+    from: Token,
+    to: Token,
+    fromValue: string,
+    frequencyType: BigNumber,
+    frequencyValue: string,
+    yieldFrom?: string,
+    yieldTo?: string,
+    isPolygonDestnantion?: boolean
+  ): Promise<TransactionResponse> {
+    const swapAndXCallParams = await this.buildXcallDepositParams(
+      from,
+      to,
+      fromValue,
+      frequencyType,
+      frequencyValue,
+      yieldFrom,
+      yieldTo,
+      isPolygonDestnantion ? 137 : 0
+    );
+
+    const txRequest = await this.connextService.prepareSwapAndXCallHelper(
+      swapAndXCallParams,
+      this.walletService.account as string
+    );
+
+    return this.providerService.sendTransaction(txRequest as TransactionRequest);
```

Comment:
is txRequest not an actual TransactionRequest?

Reply:
Type here is: `TransactionRequest | undefined`,

Reply:
then we should check for the txRequest being undefined